### DISCUSSION TITLE
Create a new form component to render HTML text values

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5859] [Feature Requests]
<img width="930" alt="image" src="https://github.com/specify/specify7/assets/37256050/2c22edfb-ef8d-4c30-90c3-a56c5b91cb86">

So things like this

```html
Sn<sup>2+</sup><sub>21</sub>O<sub>6</sub>(OH)<sub>14</sub>Cl<sub>16</sub>
```

are rendered correctly when the form definition tells it to

**Comment by maxpatiiuk** [comment]
note, executing arbitrary html is dangerous (Cross-site scripting (XSS)  and HTML injection)

the HTML code MUST be sanitized before being outputted to remove unsafe tags (`<script>`, `onerror=""`, ...)

there are libraries that can do that. fortunately, there is now also an experimental API that can do that - https://developer.mozilla.org/en-US/docs/Web/API/HTML_Sanitizer_API

**Comment by maxpatiiuk** [comment]
@grantfitzsimmons can we expect users to be able to write HTML?
or do we need a wysiwyg editor too?

**Comment by maxpatiiuk** [comment]
Related to https://github.com/specify/specify7/issues/3660

**Comment by grantfitzsimmons** [comment]
I was asking the same questions with @melton-jason yesterday,
Both about my concern about executing arbitrary HTML _and_ what kind of interface they would enter this information with.

I think we'd need to have a wysiwyg editor, but we could keep it simple at first if we make this work for chemical formulas only.


https://www.efofex.com/fxchem.php

<img width="1016" alt="image" src="https://github.com/specify/specify7/assets/37256050/6e012a3e-354a-4e59-b1c1-fcd4e04afefc">

**Comment by maxpatiiuk** [comment]
fyi: MathML is finally supported by all major browsers
there should be editors out there that support it - https://developer.mozilla.org/en-US/docs/Web/MathML/Authoring#wysiywg_editors
